nistats/tests/test_glm_reporter.py

```diff
@@ -0,0 +1,218 @@
+import os
+
+import nibabel as nib
+import numpy as np
+
+from nose import SkipTest
+
+import pandas as pd
+from nilearn.image import concat_imgs
+from nose.tools import assert_true, assert_equal
+
+from nistats._utils.testing import _write_fake_fmri_data
+from nistats.design_matrix import make_first_level_design_matrix
+from nistats.first_level_model import FirstLevelModel
+from nistats.reporting import glm_reporter as glmr
+from nistats.second_level_model import SecondLevelModel
+from nistats import datasets
+from numpy.testing import dec
+
+try:
+    import matplotlib as mpl
+except ImportError:
+    not_have_mpl = True
+else:
+    not_have_mpl = False
+
+
+@dec.skipif(not_have_mpl)
+def test_flm_fiac_test():
+    if mpl.__version__ == '1.5.1':
+        raise SkipTest('Skipping test in Matplotlib v1.5.1')
+    data = datasets.fetch_fiac_first_level()
+    fmri_img = [data['func1'], data['func2']]
+    
+    from nilearn.image import mean_img
+    mean_img_ = mean_img(fmri_img[0])
+    
+    design_files = [data['design_matrix1'], data['design_matrix2']]
+    design_matrices = [pd.DataFrame(np.load(df)['X']) for df in design_files]
+    
+    fmri_glm = FirstLevelModel(mask_img=data['mask'], minimize_memory=True)
+    fmri_glm = fmri_glm.fit(fmri_img, design_matrices=design_matrices)
+    
+    n_columns = design_matrices[0].shape[1]
+    
+    def pad_vector(contrast_, n_columns):
+        """A small routine to append zeros in contrast vectors"""
+        return np.hstack((contrast_, np.zeros(n_columns - len(contrast_))))
+    
+    contrasts = {'SStSSp_minus_DStDSp': pad_vector([1, 0, 0, -1], n_columns),
+                 'DStDSp_minus_SStSSp': pad_vector([-1, 0, 0, 1], n_columns),
+                 }
+    report = glmr.make_glm_report(fmri_glm, contrasts, bg_img=mean_img_,
+                                  roi_img=data['mask'])
+    benchmark_filepath = 'fiac_flm_report_benchmark.html'
```

Comment:
What is tested here ?

Reply:
It's an end to end integration-smoke test. If there are no errors that's good. Requires some more seasoning.